File: 2023/javascript/day_01/day_1_1.js
const {readFileSync} = require("fs")

const input = readFileSync("./input.txt", "utf-8").replace(/\r/g, "")
const codes = input.split("\n")

const numbers = []

for(let code of codes) {
    let first, last;

    for(let char of code.split("")) {
        const parsed = parseInt(char)
        if(!isNaN(parsed)) {
            if(first === undefined) first = parsed
            last = parsed
        }
    }
    numbers.push(first * 10 + last)
}

const sum = numbers.reduce((a, b) => a + b)

console.log(`The sum of all numbers is: ${sum}`)

File: 2023/python/day_01/day_1_1.py
lines = open("input.txt", "r").read().replace(r"\r", "").split("\n")

sum = 0

for line in lines:
    first = None
    last = None
    for char in line:
        if char.isnumeric():
            last = int(char)
            if first is None:
                first = int(char)
    sum += (first * 10) + last


print(f"The sum of all calibration values is: {sum}")